lib/OpenQA/WebAPI/Controller/Test.pm

```diff
@@ -90,30 +75,53 @@ sub list_ajax {
         prefetch => [qw(children parents)],
     )->all;
 
-    # need to use all as the order is too complex for a cursor
-    my $comment_count = $self->prefetch_comment_counts([map { $_->id } @jobs]);
+    my $comment_data = $self->schema->resultset('Comments')->comment_data_for_jobs(\@jobs, {bugdetails => 1});
     my @list;
     for my $job (@jobs) {
         my $job_id = $job->id;
+        my $cd     = $comment_data->{$job_id} // 0;
+        my %comment_data;
+        if ($cd) {
+            $comment_data{comments}     = $cd->{comments};
+            $comment_data{reviewed}     = $cd->{reviewed};
+            $comment_data{label}        = $cd->{label};
+            $comment_data{comment_icon} = $self->comment_icon($job_id, $comment_data{comments});
+            my $bugs = $cd->{bugs};
+            my @bugs;
+            my $details = $cd->{bugdetails};
+            for my $bugid (sort keys %$bugs) {
+                my $bug  = $details->{$bugid};
+                my $item = {
+                    id        => $bug->id,
+                    content   => $bugid,
+                    closed    => $bug->open,
+                    title     => $self->bugtitle_for($bugid, $bug),
+                    url       => $self->bugurl_for($bugid),
+                    css_class => $self->bugicon_for($bugid, $bug),
+                };
+                push @bugs, $item;
```

Comment:
Could the array be populated using `map`?

Reply:
ok, I rearranged the code to use map and saved some temp vars